File: socha-sdk/src/server-api/sc/networking/clients/IControllableGame.java
package sc.networking.clients;

import sc.shared.GameResult;

public interface IControllableGame extends IPollsUpdates
{
	public void next();

	public void previous();

	public void pause();

	public void unpause();

	public boolean hasNext();

	public boolean hasPrevious();

	public boolean isPaused();
	
	public void cancel();

	public Object getCurrentState();
	
	public Object getCurrentError();

	public boolean isAtEnd();

	public boolean isAtStart();

	public void goToFirst();

	public void goToLast();

	public boolean canTogglePause();
	
	public GameResult getResult();

	public boolean isGameOver();

	public boolean isReplay();
}

File: sdk/src/server-api/sc/networking/clients/IControllableGame.kt
package sc.networking.clients

import sc.shared.GameResult

interface IControllableGame {
    fun next()
    fun hasNext(): Boolean
    
    fun previous()
    fun pause()
    fun unpause()
    fun hasPrevious(): Boolean
    val isPaused: Boolean
    fun cancel()
    val currentState: Any?
    val currentError: Any?
    val isAtEnd: Boolean
    val isAtStart: Boolean
    fun goToFirst()
    fun goToLast()
    fun canTogglePause(): Boolean
    val result: GameResult?
    val isGameOver: Boolean
    val isReplay: Boolean
}